examples/Braccio_Learn_and_Repeat/Braccio_Learn_and_Repeat.ino

```diff
@@ -0,0 +1,165 @@
+        }
+        break;
+      case 1:
+        btnm_map[0] = "LEARN"; // reset the label of the first button back to "LEARN"
+        if (txt == "REPEAT"){
+        state = REPEAT;
+        btnm_map[2] = "STOP"; // change the label of the second button to "STOP"
+        Braccio.engage();
+        lv_btnmatrix_set_btn_ctrl(btnm, 1, LV_BTNMATRIX_CTRL_CHECKED);
+        Serial.println("REPEAT");
+        }
+        else if (txt=="STOP"){
+          state = ZERO_POSITION;
+          Braccio.engage(); // enable the steppers so that the braccio stands still
+          lv_btnmatrix_set_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKED);
+          btnm_map[2] = "REPEAT"; // reset the label of the first button back to "REPEAT"
+        }
+        
+        break;
+      
+      default:
+        state = ZERO_POSITION;
+        btnm_map[0] = "LEARN"; // reset the label of the first button back to "LEARN"
+        btnm_map[2] = "REPEAT"; // reset the label of the first button back to "REPEAT"
+        Braccio.engage();
+        delay(500);
+        Braccio.moveTo(homePos[0], homePos[1], homePos[2], homePos[3], homePos[4], homePos[5]);
+        lv_btnmatrix_set_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKED);
+        Serial.println("ZERO_POSITION");
+        break;
+    }
+  }
+}
+
+void mainMenu() {
+  static lv_style_t style_focus;
+  lv_style_init(&style_focus);
+  lv_style_set_outline_color(&style_focus, lv_color_hex(COLOR_ORANGE));
+  lv_style_set_outline_width(&style_focus, 4);
+
+  static lv_style_t style_btn;
+  lv_style_init(&style_btn);
+  lv_style_set_bg_color(&style_btn, lv_color_hex(COLOR_LIGHT_TEAL));
+  lv_style_set_text_color(&style_btn, lv_color_white());
+
+  btnm = lv_btnmatrix_create(lv_scr_act());
+  lv_obj_set_size(btnm, 240, 240);
+  lv_btnmatrix_set_map(btnm, btnm_map);
+  lv_obj_align(btnm, LV_ALIGN_CENTER, 0, 0);
+
+  lv_obj_add_style(btnm, &style_btn, LV_PART_ITEMS);
+  lv_obj_add_style(btnm, &style_focus, LV_PART_ITEMS | LV_STATE_FOCUS_KEY);
+
+  lv_btnmatrix_set_btn_ctrl(btnm, 0, LV_BTNMATRIX_CTRL_CHECKABLE);
+  lv_btnmatrix_set_btn_ctrl(btnm, 1, LV_BTNMATRIX_CTRL_DISABLED);
+  lv_btnmatrix_set_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKABLE);
+
+  lv_btnmatrix_set_one_checked(btnm, true);
+  lv_btnmatrix_set_selected_btn(btnm, 0);
+  lv_btnmatrix_set_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKED);
+
+  counter = lv_label_create(btnm);
+  lv_label_set_text_fmt(counter, "Counter: %d" , 0);
+  lv_obj_align(counter, LV_ALIGN_CENTER, 0, 80);
+
+  lv_obj_add_event_cb(btnm, eventHandlerMenu, LV_EVENT_ALL, NULL);
+
+  Braccio.connectJoystickTo(btnm);
+}
+
+void setup() {
+  Braccio.begin(mainMenu);
+  delay(500);
+
+  Braccio.moveTo(homePos[0], homePos[1], homePos[2], homePos[3], homePos[4], homePos[5]);
+  delay(500);
+  
+  Serial.begin(115200);
+  Serial.println("Replicate a movement");
+}
+
+void loop() {
+  if (state == LEARN) {
+    Braccio.positions(idx);
+    idx += 6;
```

Comment:
There is no limitation in the code regards the maximum amount of joint angles that can be captured during the learning phase. If more than 10.000 joint positions are captured (1 capture = 6 joint positions, 1 capture every 100 ms -> after 166 seconds of training) we have an array overflow which will lead to stack corruption and a hard fault.

In order to prevent this this code should be extended to automatically leave learning state when the maximum amount of joint positions have been captured.

Reply:
I've tested `Braccio_Learn_and_Repeat.ino`, and it ran correctly. I faced the same issue that Alex reported; if more than 10.000 joint positions are captured, we have an array overflow, leading to stack corruption. The firmware stopped working. Thus, it was necessary to restart the Arduino Board.

I liked that the use of Flash memory is encapsulated in the header file.

Regarding the suggestion to make the code identical to both `Braccio_Learn_and_Repeat.ino` and `01_Braccio_learning_mode.ino`, please let me know. With the new feature, I will already have to update the content of project 03. So, I do only one task for all updates. @clodpheasant @GDJose

Reply:
Also, if you decide to make both codes identical, it will be necessary to change some videos and images in project 03.

Reply:
I would make them identical. Otherwise in n months the next person comes along scratching their head. Alternative would be to delete on of the two. @clodpheasant please advise how to move forward.

Reply:
Sure thing, these sketches have to be identical. And @Esquirio yes we need to change the content on the platform also to be aligned with this new release.

Reply:
I've aligned the two sketches.

However, there will still be some changes in order to fix the array overflow.

BUT: not in this PR :wink: 

I'll merge this PR so that we then can iterate in smaller PRs on the other issues.

For @Esquirio this means: do not update content just yet.